test/TestAddressSetRemove.sol

```diff
@@ -0,0 +1,146 @@
+    }
+
+    function testRemoveDuplicate() public {
+        shouldAdd(ADDRESS_1);
+        shouldRemove(ADDRESS_1);
+        shouldNotRemove(ADDRESS_1);
+
+        set.shouldBeEmpty();
+        set.shouldNotContain(ADDRESS_1);
+    }
+
+    function testRemoveFirst() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        shouldAdd(ADDRESS_3);
+        shouldRemove(ADDRESS_1);
+
+        set.shouldEqual([ADDRESS_2, ADDRESS_3]);
+        set.shouldNotContain(ADDRESS_1);
+    }
+
+    function testRemoveMiddle() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        shouldAdd(ADDRESS_3);
+        shouldRemove(ADDRESS_2);
+
+        set.shouldEqual([ADDRESS_1, ADDRESS_3]);
+        set.shouldNotContain(ADDRESS_2);
+    }
+
+    function testRemoveLast() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        shouldAdd(ADDRESS_3);
+        shouldRemove(ADDRESS_3);
+
+        set.shouldEqual([ADDRESS_1, ADDRESS_2]);
+        set.shouldNotContain(ADDRESS_3);
+    }
+
+    function testRemoveAll() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        shouldAdd(ADDRESS_3);
+        shouldRemove(ADDRESS_3);
+        shouldRemove(ADDRESS_1);
+        shouldRemove(ADDRESS_2);
+
+        set.shouldBeEmpty();
+        set.shouldNotContain(ADDRESS_1);
+        set.shouldNotContain(ADDRESS_2);
+        set.shouldNotContain(ADDRESS_3);
+    }
+
+    function testRemoveFirstAndAdd() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        shouldRemove(ADDRESS_1);
+        shouldAdd(ADDRESS_3);
+
+        set.shouldEqual([ADDRESS_2, ADDRESS_3]);
+        set.shouldNotContain(ADDRESS_1);
+    }
+
+    function testRemoveMiddleAndAdd() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        shouldAdd(ADDRESS_3);
+
+        shouldRemove(ADDRESS_2);
+        shouldAdd(ADDRESS_2);
+
+        set.shouldEqual([ADDRESS_1, ADDRESS_3, ADDRESS_2]);
+    }
+
+    function testRemoveLastAndAdd() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        shouldRemove(ADDRESS_2);
+        shouldAdd(ADDRESS_3);
+
+        set.shouldEqual([ADDRESS_1, ADDRESS_3]);
+        set.shouldNotContain(ADDRESS_2);
+    }
+
+    function testClear() public {
+        shouldAdd(ADDRESS_1);
+        shouldAdd(ADDRESS_2);
+        set.clear();
+
+        set.shouldBeEmpty();
+        set.shouldNotContain(ADDRESS_1);
+        set.shouldNotContain(ADDRESS_2);
+    }
+
+    function shouldAdd(address addr) private {
+        bool added = set.add(addr);
+        Assert.isTrue(added, "Should add");
+    }
+
+    function shouldRemove(address addr) private {
+        bool removed = set.remove(addr);
+        Assert.isTrue(removed, "Should remove");
+    }
+
+    function shouldNotRemove(address addr) private {
+        bool removed = set.remove(addr);
+        Assert.isFalse(removed, "Should not remove");
+    }
+}
```

Comment:
same as with previous `shoulds`, 
`shouldRemove` -> `remove`
`shouldNotRemove` -> `removeMissing`
?

Reply:
Changed to `remove` and `removeNotExisting`